src/utils/hooks/api/loans/lend-and-borrow-info.tsx

```diff
@@ -0,0 +1,243 @@
+): Big => {
+  if (!supplyAmountUSD.gt(0)) {
+    return new Big(0);
+  }
+
+  const totalLendApy = lendPositions.reduce((total, position) => {
+    const { lendApy, lendReward } = assets[position.currency.ticker];
+    const rewardsApy = getSubsidyRewardApy(position.currency, lendReward, prices);
+    const positionApy = lendApy.add(rewardsApy || 0);
+    const positionUSDValue = convertMonetaryAmountToValueInUSD(
+      position.amount,
+      getTokenPrice(prices, position.currency.ticker)?.usd
+    );
+
+    return positionUSDValue ? total.add(positionApy.mul(positionUSDValue)) : total;
+  }, new Big(0));
+
+  const totalBorrowApy = borrowPositions.reduce((total, position) => {
+    const { borrowApy, borrowReward } = assets[position.currency.ticker];
+    const rewardsApy = getSubsidyRewardApy(position.currency, borrowReward, prices);
+    const positionApy = borrowApy.sub(rewardsApy || 0);
+    const positionUSDValue = convertMonetaryAmountToValueInUSD(
+      position.amount,
+      getTokenPrice(prices, position.currency.ticker)?.usd
+    );
+
+    return positionUSDValue ? total.add(positionApy.mul(positionUSDValue)) : total;
+  }, new Big(0));
+
+  return totalLendApy.sub(totalBorrowApy).div(supplyAmountUSD);
+};
+
+const getAccountPositionsStats = (
+  assets: TickerToData<LoanAsset>,
+  lendPositions: LendPosition[],
+  borrowPositions: BorrowPosition[],
+  subsidyRewards: MonetaryAmount<CurrencyExt>,
+  prices: Prices
+): AccountPositionsStatisticsData => {
+  const supplyAmountUSD = getPositionsSumOfFieldsInUSD('amount', lendPositions, prices);
+
+  const borrowAmountUSD = getPositionsSumOfFieldsInUSD('amount', borrowPositions, prices);
+
+  const collateralLendPositions = lendPositions.filter(({ isCollateral }) => isCollateral);
+
+  const collateralPositionsWithAppliedCollateralThreshold = collateralLendPositions.map(
+    ({ amount, currency, ...rest }) => ({
+      // MEMO: compute total value based on collateral threshold (not full lend amount value)
+      amount: amount.mul(assets[currency.ticker].collateralThreshold),
+      currency,
+      ...rest
+    })
+  );
+
+  const collateralPositionsWithAppliedLiquidationThreshold = collateralLendPositions.map(
+    ({ amount, currency, ...rest }) => ({
+      // MEMO: compute total value based on collateral threshold (not full lend amount value)
+      amount: amount.mul(assets[currency.ticker].liquidationThreshold),
+      currency,
+      ...rest
+    })
+  );
+
+  const collateralAmountUSD = getPositionsSumOfFieldsInUSD(
+    'amount',
+    collateralPositionsWithAppliedCollateralThreshold,
+    prices
+  );
+  const liquidationAmountUSD = getPositionsSumOfFieldsInUSD(
+    'amount',
+    collateralPositionsWithAppliedLiquidationThreshold,
+    prices
+  );
+
+  const collateralizedAmountUSD = getPositionsSumOfFieldsInUSD('amount', collateralLendPositions, prices);
+  const earnedDeptAmountUSD = getPositionsSumOfFieldsInUSD('accumulatedDebt', borrowPositions, prices);
```

Comment:
These should be refactored using `totalLentBtc`, `totalCollateralBtc`, `totalBorrowedBtc`, `borrowLimitBtc` from `LoanCollateralInfo`. Check `useLoanInfo` where you can get `LoanCollateralInfo`.